pkg/cloudprovider/providers/gce/gce_addresses_fakes.go

```diff
@@ -34,14 +36,33 @@ type FakeCloudAddressService struct {
 	addrsByRegionAndName map[string]map[string]*compute.Address
 }
 
+// FakeCloudAddressService Implements CloudAddressService
+var _ CloudAddressService = &FakeCloudAddressService{}
+
 func NewFakeCloudAddressService() *FakeCloudAddressService {
 	return &FakeCloudAddressService{
 		reservedAddrs:        make(map[string]bool),
 		addrsByRegionAndName: make(map[string]map[string]*compute.Address),
 	}
 }
 
+func (cas *FakeCloudAddressService) SetRegionalAddresses(region string, ipList map[string]string) {
+	cas.Lock()
+	defer cas.Unlock()
+	if _, exists := cas.addrsByRegionAndName[region]; !exists {
+		cas.addrsByRegionAndName[region] = make(map[string]*compute.Address)
+	}
+
+	regionStore := cas.addrsByRegionAndName[region]
+	for name, ip := range ipList {
+		cas.reservedAddrs[ip] = true
+		regionStore[name] = &compute.Address{Name: name, Address: ip}
+	}
+}
```

Comment:
Recommend checking that the IP/address-name doesn't exist before creating the map entries and returning a potential error. Mistakes in unit tests do happen and would be nice to catch them then.